src/sched/mod.rs

```diff
@@ -24,14 +65,21 @@ pub mod waker;
 /// 3. If the selected process is the same as the currently running one, no
 ///    switch occurs.
 /// 4. If a new process is selected, it handles the state transitions (`Running`
-///    -&gt; `Runnable` for the old task, `Runnable` -&gt; `Running` for the new task)
-///    and performs the architecture-specific context switch.
+///   > `Runnable` for the old task, `Runnable` > `Running` for the new task)
+///   > and performs the architecture-specific context switch.
 ///
 /// # Returns
 ///
 /// Nothing, but the CPU context will be set to the next runnable task. See
 /// `userspace_return` for how this is invoked.
 fn schedule() {
+    if SCHED_STATE.try_borrow_mut().is_none() {
```

Comment:
This is my "fix" for that, but it just returns. Sometimes this happens instantly, causing the idle task to be executed ad-nauseam.